termsaverlib/screen/base/imagereader.py

```diff
@@ -0,0 +1,347 @@
+
+            * path: defines the path from where this screen should scan
+              for files
+        """
+        ScreenBase.__init__(self, name, description, cli_opts)
+        # define default cli options, if none are informed
+        if not cli_opts:
+            self.cli_opts = {
+                             'opts': 'hd:p:',
+                             'long_opts': ['help', 'delay=', 'path='],
+            }
+        self.image_delay = delay
+        self.path = path
+        self.cleanup_per_cycle = False
+        self.options = {'wide':2}
+
+    def _run_cycle(self):
+        """
+        Executes a \"cycle\" of this screen.
+            * The concept of \"cycle\" is no longer accurate, and is misleading.
+              this function will not return.
+        New threaded implementation:
+            * Checks if self.path is a valid path, using `os.path.exists`
+            * Assigns a new Queue to `queue_of_valid_files`
+            * Appends a new `FileScannerThread` object to a list of threads
+            * `start()`s the `FileScannerThread`
+                * `FileScannerThread` will put paths in the queue as valid
+                   file paths are found
+            * `clear_screen()`s
+            * Gets a file from `queue_of_valid_files`, removing item from queue
+            * While nextFile (empty sequences are false)
+                * As long as there is something in the queue - that is, as long
+                  as `queue.queue.get()` is able to get an object from (the)
+                  `queue_of_valid_files`, this test evaluates True.
+                * I imagine that this behaves unpredictably given a computer
+                  with __REALLY__ slow I/O
+            * Opens `nextFile` with handle-auto-closing `with` statement and
+              `typing_print()`s it
+            * Clears screen if `self.cleanup_per_file`
+            * Puts `nextFile` ON the queue
+                * Because `queue_of_valid_files.get()` REMOVES a file path
+                  from the queue, `_run_cycle()` will never reach that path
+                  again, and eventually will exhaust the queue
+                  (failing silently, with a blank screen)
+                    * A static blank screen is the antithesis of a screensaver
+                * Therefore, `queue_of_valid_files.put(nextFile)` puts the file
+                  path at the last spot in the queue
+            * Finally, another call to `queue_of_valid_files.get()` sets up
+              the next iteration in the while loop.
+        """
+        # validate path
+        if not os.path.exists(self.path):
+            raise exception.PathNotFoundException(self.path)
+
+        queue_of_valid_files = queue.Queue()
+
+        threads = [ImageReaderBase.FileScannerThread(self, queue_of_valid_files, self.path)]
+        threads[-1].daemon = True
+        threads[-1].start()
+        #self.clear_screen() hides any error message produced before it!
```

Comment:
Can you remove commented code, unless you need to keep there for future use?

Reply:
That's the plan! As I've stated in all of these PRs they're drafts, not meant to be taken as ready for merging unless we agree otherwise. They're more for you to test out functionality and make sure it's something we still want to go ahead with.